File: SigreAppMobile/Hermes/common-ui/src/main/java/pe/com/hermes/common/ui/LoadingDialog.java
package pe.com.hermes.common.ui;

import android.app.Activity;
import android.app.Dialog;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import pe.com.hermes.common.databinding.DialogLoadingBinding;

/**
 * Dialogo de carga (spinner + mensaje) con show/dismiss seguros — equivalente
 * moderno de SafeProgress.java de FastSales.
 */
public final class LoadingDialog {

    private LoadingDialog() {
    }

    public static Dialog mostrar(Activity activity, String mensaje) {
        if (activity.isFinishing() || activity.isDestroyed()) return null;
        DialogLoadingBinding binding = DialogLoadingBinding.inflate(activity.getLayoutInflater());
        binding.tvMensaje.setText(mensaje);
        return new MaterialAlertDialogBuilder(activity)
                .setView(binding.getRoot())
                .setCancelable(false)
                .show();
    }

    public static Dialog mostrar(Activity activity) {
        return mostrar(activity, "Cargando…");
    }

    public static void cerrar(Activity activity, Dialog dialog) {
        if (dialog == null) return;
        if (activity.isFinishing() || activity.isDestroyed()) return;
        if (dialog.isShowing()) dialog.dismiss();
    }
}

File: SigreAppMobile/Hermes/common-ui/src/main/java/pe/com/hermes/common/ui/LoadingDialog.kt
package pe.com.hermes.common.ui

import android.app.Activity
import android.app.Dialog
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import pe.com.hermes.common.databinding.DialogLoadingBinding

/**
 * Diálogo de carga (spinner + mensaje) con show/dismiss seguros — equivalente
 * moderno de SafeProgress.java de FastSales (que envolvía el hoy-deprecado
 * ProgressDialog) y del uso que le daba spots-dialog en las pantallas de
 * Compras. Verifica que la Activity no esté finishing/destroyed antes de
 * mostrar o cerrar, para evitar el WindowLeaked/IllegalArgumentException
 * clásico de mostrar un diálogo sobre una Activity ya cerrada.
 */
object LoadingDialog {

    fun mostrar(activity: Activity, mensaje: String = "Cargando…"): Dialog? {
        if (activity.isFinishing || activity.isDestroyed) return null
        val binding = DialogLoadingBinding.inflate(activity.layoutInflater)
        binding.tvMensaje.text = mensaje
        return MaterialAlertDialogBuilder(activity)
            .setView(binding.root)
            .setCancelable(false)
            .show()
    }

    fun cerrar(activity: Activity, dialog: Dialog?) {
        if (dialog == null) return
        if (activity.isFinishing || activity.isDestroyed) return
        if (dialog.isShowing) dialog.dismiss()
    }
}
